cloud/amazon/sts_assume_role.py

```diff
@@ -0,0 +1,166 @@
+'''
+
+EXAMPLES = '''
+# Basic example of assuming a role
+tasks:
+- name: assume a role in account 123456789012
+  sts_assume_role: role_arn="arn:aws:iam::123456789012:role/someRole" session_name="someRoleSession"
+
+- name: display temporary credentials
+  debug: "temporary credentials for the assumed role are {{ ansible_temp_credentials }}"
+  
+- name: use temporary credentials for tagging an instance in account 123456789012 
+  ec2_tag: resource=i-xyzxyz01 region=us-west-1 state=present
+    args:
+      aws_access_key: "{{ ansible_temp_credentials.access_key }}"
+      aws_secret_key: "{{ ansible_temp_credentials.secret_key }}"
+      security_token: "{{ ansible_temp_credentials.session_token }}"
+    
+      tags:
+        Test: value
+'''
+
+import sys
+import time
+
+try:
+    import boto.sts
+    
+except ImportError:
+    print "failed=True msg='boto required for this module'"
+    sys.exit(1)
+
+def sts_connect(module):
+
+    """ Return an STS connection"""
+
+    region, ec2_url, boto_params = get_aws_connection_info(module)
+
+    # If we have a region specified, connect to its endpoint.
+    if region:
+        try:
+            sts = connect_to_aws(boto.sts, region, **boto_params)
+        except boto.exception.NoAuthHandlerFound, e:
+            module.fail_json(msg=str(e))
+    # Otherwise, no region so we fallback to connect_sts method
+    else:
+        try:
+            sts = boto.connect_sts(**boto_params)
+        except boto.exception.NoAuthHandlerFound, e:
+            module.fail_json(msg=str(e))
+
+
+    return sts
+
+def assumeRole():
+    data = sts.assume_role()
+    return data
+
+def main():
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(dict(
+            role_arn = dict(required=True),
+            role_session_name = dict(required=True),
+            duraction_seconds = dict(),
+            external_id = dict(),
+            policy = dict(),
+            mfa_serial_number = dict(),
+            mfa_token = dict(),
+        )
+    )
+    module = AnsibleModule(argument_spec=argument_spec)
+
+    role_arn = module.params.get('role_arn')
+    role_session_name = module.params.get('role_session_name')
+    policy = module.params.get('policy')
+    duraction_seconds = module.params.get('duraction_seconds')
+    external_id = module.params.get('external_id')
+    mfa_serial_number = module.params.get('mfa_serial_number')
+    mfa_token = module.params.get('mfa_token')
+  
+    sts = sts_connect(module)
+    
+    temp_credentials = {}
+    
+    try:
+        temp_credentials = sts.assume_role(role_arn, role_session_name, policy, duraction_seconds,
+                                           external_id, mfa_serial_number, mfa_token).credentials.__dict__
+    except boto.exception.BotoServerError, e:
+        module.fail_json(msg='Unable to assume role {0}, error: {1}'.format(role_arn, e))
+    result = dict(changed=False, ansible_facts=dict(ansible_temp_credentials=temp_credentials))

```

Comment:
@bekelchik I agree with @bpennypacker  comments. If you can take care of the items both of us made comments on, I would be in favor of merging this.